### PR TITLE
Replace manual miniredis setup with statestore testing utility code

### PR DESCRIPTION
**Pull request by yfei1**
Expose a statestore util function for testing purpose and replace repetitive miniredis manual setup this function.

### CODE REVIEW

File: internal/app/mmlogic/mmlogic_test.go [hunk 1]
```diff
@@ -17,12 +17,30 @@ package mmlogic
...
 	"google.golang.org/grpc"
 	"open-match.dev/open-match/internal/pb"
 	"open-match.dev/open-match/internal/rpc"
 	rpcTesting "open-match.dev/open-match/internal/rpc/testing"
+	statestoreTesting "open-match.dev/open-match/internal/statestore/testing"
 )
 
+func TestQueryTicketsBasic(t *testing.T) {
+	assert := assert.New(t)
+	vp := viper.New()
```
Comment (Laremere) [inline internal/app/mmlogic/mmlogic_test.go]: why is this "vp" instead of "cfg"?

File: internal/app/frontend/frontend_test.go [hunk 2]
```diff
@@ -76,7 +75,12 @@ func validateDelete(t *testing.T, fe *frontendService, id string) {
 // TestFrontendService tests creating, getting and deleting a ticket using Frontend service.
 func TestFrontendService(t *testing.T) {
 	assert := assert.New(t)
-	cfg := createStore(t)
+	vp := viper.New()
+	vp.Set("playerIndices", []string{"testindex1", "testindex2"})
```
Comment (Laremere) [inline internal/app/frontend/frontend_test.go]: This is also set by NewRedisForTesting.

File: internal/app/mmlogic/mmlogic_test.go (hunk 1 above)
Comment (yfei1) [reply internal/app/mmlogic/mmlogic_test.go]: because cfg is used at Line 34.

File: internal/statestore/testing/public.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import (
+	"time"
+
+	"github.com/alicebob/miniredis"
+	"github.com/spf13/viper"
+	"open-match.dev/open-match/internal/config"
+)
+
+// NewRedisForTesting create a new in-memory miniredis for testing
+func NewRedisForTesting(cfg *viper.Viper) (config.View, func(), error) {
```
Comment (sawagh) [inline internal/statestore/testing/public.go]: NewTestStateStorage would be more apt. Might be best to abstract the fact that there is redis under the covers - from any outside components using this.

Comment (sawagh) [inline internal/statestore/testing/public.go]: This should not expose the closer - any component using this cfg to build a StateStore object should be calling .Close() on the public state store method. 

Do consider not reusing this helper for redis_test.go.

**Comment by yfei1** [comment]
Closed as #462 resolved most of the issues in this PR.

**Pull request closed without merging**